dpnp/dpnp_algo/dpnp_elementwise_common.py

```diff
@@ -24,89 +24,546 @@
+            A callback to register in UnaryElementwiseFunc class of
+            dpctl.tensor
+            """
+
+            if depends is None:
+                depends = []
+
+            if mkl_fn_to_call is not None and mkl_fn_to_call(
+                sycl_queue, src, dst
+            ):
+                # call pybind11 extension for unary function from OneMKL VM
+                return mkl_impl_fn(sycl_queue, src, dst, depends)
+            return unary_dp_impl_fn(src, dst, sycl_queue, depends)
+
+        super().__init__(
+            name,
+            result_type_resolver_fn,
+            _call_func,
+            docs,
+            acceptance_fn=acceptance_fn,
+        )
+        self.__name__ = "DPNPUnaryFunc"
+        self.origin_fn = origin_fn
+
+    def __call__(
+        self,
+        x,
+        out=None,
+        where=True,
+        order="K",
+        dtype=None,
+        subok=True,
+        **kwargs,
+    ):
+        if kwargs:
+            if self.origin_fn is None:
+                raise NotImplementedError(
+                    f"Requested function={self.name_} with kwargs={kwargs} "
+                    "isn't currently supported."
+                )
+        elif where is not True:
+            if self.origin_fn is None:
+                raise NotImplementedError(
+                    f"Requested function={self.name_} with where={where} "
+                    "isn't currently supported."
+                )
+        elif dtype is not None:
+            if self.origin_fn is None:
+                raise NotImplementedError(
+                    f"Requested function={self.name_} with dtype={dtype} "
+                    "isn't currently supported."
+                )
+        elif subok is not True:
+            if self.origin_fn is None:
+                raise NotImplementedError(
+                    f"Requested function={self.name_} with subok={subok} "
+                    "isn't currently supported."
+                )
+        elif dpnp.isscalar(x):
+            # input has to be an array
+            if self.origin_fn is None:
+                raise NotImplementedError(
+                    f"Requested function={self.name_} with args={x} "
+                    "isn't currently supported."
+                )
+        else:
+            if order in "afkcAFKC":
+                order = order.upper()
+            elif order is None:
+                order = "K"
+            else:
+                raise ValueError(
+                    "order must be one of 'C', 'F', 'A', or 'K' "
+                    f"(got '{order}')"
+                )
+            x_usm = dpnp.get_usm_ndarray(x)
+            out_usm = None if out is None else dpnp.get_usm_ndarray(out)
+            res_usm = super().__call__(x_usm, out=out_usm, order=order)
+            if out is not None and isinstance(out, dpnp_array):
+                return out
+            return dpnp_array._create_from_usm_ndarray(res_usm)
+
+        return call_origin(
```

Comment:
Is there any simple way to get rid of `call_origin` completely for all elementwise functions?

Reply:
Elementwise functions may raise NotImplementedError for all unsupported parameters

Reply:
Yes, I think we should be that way.